src/jit/hwintrinsicxarch.cpp

```diff
@@ -537,6 +694,8 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
                     strip(info.compCompHnd->getArgType(sig, info.compCompHnd->getArgNext(argList), &argClass)));
                 op2 = getArgForHWIntrinsic(argType, argClass);
 
+                op2 = addRangeCheck(intrinsic, op2, mustExpand);
```

Comment:
Is this getting emitted even for the full range intrinsics?

Reply:
I will change here to handle non-full-range intrinsics.

Reply:
No, I meant I will change it in this PR 😄

Reply:
Fixed in the function-body of `addRangeCheck`.

Reply:
> Fixed in the function-body of addRangeCheck.

I'd suggest renaming to `addRangeCheckIfNeeded`.

Reply:
> I'd suggest renaming to addRangeCheckIfNeeded.

Good point! will change.